sockeye/vocab.py

```diff
@@ -253,36 +263,38 @@ def load_target_vocabs(folder: str) -> List[Vocab]:
     return [vocab_from_json(os.path.join(folder, fname)) for fname in _get_sorted_target_vocab_fnames(folder)]
 
 
-def load_or_create_vocab(data: str, vocab_path: Optional[str], num_words: int, word_min_count: int,
-                         pad_to_multiple_of: Optional[int] = None) -> Vocab:
+def load_or_create_vocab(data: Tuple[str], vocab_path: Optional[str], num_words: int, word_min_count: int,
+                         pad_to_multiple_of: Optional[int] = None, mapper: Callable = map) -> Vocab:
     """
     If the vocabulary path is defined, the vocabulary is loaded from the path.
     Otherwise, it is built from the data file. No writing to disk occurs.
+
+    :param data: Tuple of file paths for each shard.
     """
     if vocab_path is None:
-        return build_from_paths(paths=[data], num_words=num_words, min_count=word_min_count,
-                                pad_to_multiple_of=pad_to_multiple_of)
+        return build_from_shards(paths=data, num_words=num_words, min_count=word_min_count,
+                                 pad_to_multiple_of=pad_to_multiple_of, mapper=mapper)
     else:
         return vocab_from_json(vocab_path)
 
-
-def load_or_create_vocabs(source_paths: List[str],
-                          target_paths: List[str],
+def load_or_create_vocabs(shard_source_paths: Iterable[Iterable[str]],
+                          shard_target_paths: Iterable[Iterable[str]],
                           source_vocab_paths: List[Optional[str]],
                           source_factor_vocab_same_as_source: List[bool],
                           target_vocab_paths: List[Optional[str]],
                           target_factor_vocab_same_as_target: List[bool],
                           shared_vocab: bool,
                           num_words_source: Optional[int], word_min_count_source: int,
                           num_words_target: Optional[int], word_min_count_target: int,
-                          pad_to_multiple_of: Optional[int] = None) -> Tuple[List[Vocab], List[Vocab]]:
+                          pad_to_multiple_of: Optional[int] = None,
+                          mapper: Callable = map) -> Tuple[List[Vocab], List[Vocab]]:
     """
     Returns vocabularies for source files (including factors) and target files (including factors.
     If the respective vocabulary paths are not None, the vocabulary is read from the path and returned.
     Otherwise, it is built from the support and saved to the path.
 
-    :param source_paths: The path to the source text (and optional token-parallel factor files).
-    :param target_paths: The path to the target text (and optional token-parallel factor files).
+    :param shard_source_paths: List of shards of list paths to the source text (and optional token-parallel factor files).
```

Comment:
The first index is the shard and the second index is the factor, correct? so list[shard][factor]?

Reply:
Yes